spec/lib/spree/shipment_notice_spec.rb

```diff
@@ -45,6 +45,43 @@
           expect(notice.error).to be_present
         end
       end
+
+      context 'oustanding balance and capture is not active' do
+        before do
+          Spree::Config.shipstation_capture_at_notification = false
+          expect(shipment).to receive_message_chain(:reload, :ship!)
+          expect(shipment).to receive(:touch).with(:shipped_at)
+          expect(order).to receive(:update!)
+        end
+
+        it 'payments are not captured' do
+          expect(order).to_not receive(:capture_payments!)
+          expect(notice.apply).to eq(true)
+        end
+      end
+
+      context 'oustanding balance and capture is active' do
+        before do
+          Spree::Config.shipstation_capture_at_notification = true
+          allow(order).to receive(:outstanding_balance?).and_return(true)
+          allow(order).to receive(:capture_payments!).and_return(true)
+          expect(shipment).to receive_message_chain(:reload, :ship!)
+          expect(shipment).to receive(:touch).with(:shipped_at)
+          expect(order).to receive(:update!)
+        end
```

Comment:
I don't like those do it all methods. See how complicate it is to write a test? 

I don't have a better suggestion for you, but it will be better if there's another place to call this method from. Is it not possible to attach it to the same thing that calls `ShipmentNotice#update`?